apps/studio/src/utils/sitemap.ts

```diff
@@ -14,35 +20,69 @@ type ResourceDto = Omit<
   thumbnail?: string
 }
 
+const getResourcesWithFullPermalink = (
+  resources: ResourceDto[],
+  parent: ResourceDto,
+): ResourceDto[] => {
+  const children = resources
+    .filter((resources) => {
+      return parent.type === ResourceType.RootPage
+        ? resources.parentId === null
+        : resources.parentId === parent.id
+    })
+    .map((child) => {
+      return {
+        ...child,
+        permalink: `${parent.type === ResourceType.RootPage ? "" : parent.permalink}/${child.permalink}`,
+      }
+    })
+
+  return [
+    ...children,
+    ...children.flatMap((child) =>
+      getResourcesWithFullPermalink(resources, child),
+    ),
+  ]
+}
+
 const getSitemapTreeFromArray = (
   resources: ResourceDto[],
   parentId: string | null,
-  path: string,
 ): IsomerSitemap[] | undefined => {
   if (resources.length === 0) {
     return undefined
   }
 
   // Get the immediate children of the resource with the given parent ID
-  const children = resources.filter((resource) => {
-    if (parentId === null) {
+  const children = resources
+    .filter((resource) => {
+      const hasPageChildren = resources.some((possibleChild) => {
```

Comment:
minor nit: should be `hasPageDescendants`, children here seem to imply 1 level down only

Reply:
sounds good, lemme pudate